staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go

```diff
@@ -3741,13 +3741,42 @@ func describeNetworkPolicySpec(nps networkingv1.NetworkPolicySpec, w PrefixWrite
 	} else {
 		w.Write(LEVEL_2, "%s\n", metav1.FormatLabelSelector(&nps.PodSelector))
 	}
-	w.Write(LEVEL_1, "Allowing ingress traffic:\n")
-	printNetworkPolicySpecIngressFrom(nps.Ingress, "    ", w)
-	w.Write(LEVEL_1, "Allowing egress traffic:\n")
-	printNetworkPolicySpecEgressTo(nps.Egress, "    ", w)
+
+	ingressEnabled, egressEnabled := getPolicyType(nps)
+	if ingressEnabled {
+		w.Write(LEVEL_1, "Allowing ingress traffic:\n")
+		printNetworkPolicySpecIngressFrom(nps.Ingress, "    ", w)
+	} else {
+		w.Write(LEVEL_1, "Allowing all ingress traffic\n")
+	}
+	if egressEnabled {
+		w.Write(LEVEL_1, "Allowing egress traffic:\n")
+		printNetworkPolicySpecEgressTo(nps.Egress, "    ", w)
+	} else {
+		w.Write(LEVEL_1, "Allowing all egress traffic\n")
+
+	}
 	w.Write(LEVEL_1, "Policy Types: %v\n", policyTypesToString(nps.PolicyTypes))
 }
 
+func getPolicyType(nps networkingv1.NetworkPolicySpec) (bool, bool) {
+	if len(nps.PolicyTypes) == 0 {
```

Comment:
Is there any situation when creating a Network Policy this situation be true? As an example, creating a rule but not specifying its type will generate a ``nps.PolicyTypes`` with at least one object, right?

Reply:
I don't believe so! The policy type is often not set, and defaults in that case to ingress at the very least, and additionally egress if an egress rule is present

Reply:
So I think that only doing the ``for`` checking if this is an Ingress and/or Egress policy is enough

Reply:
I don't think so